.github/workflows/docker.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Push Docker Image (version tags)
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+-*'
+    branches:
+      - development
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - name: Extract Tag Name
+      run: echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+
+    - name: Build and push Docker image
+      run: |
+        docker-compose build --tag nucypher/nucypher:$VERSION .
```

Comment:
Instead of VERSION, I think you can just use `github.ref_name` assuming the filter is only for tags - https://docs.github.com/en/actions/learn-github-actions/contexts#github-context.